packages/suite/src/actions/wallet/stakeActions.ts

```diff
@@ -0,0 +1,213 @@
+            if (metadata.enabled) {
+                const { precomposedForm } = getState().wallet.stake;
+                const outputsPermutation = precomposedTx?.outputsPermutation;
+
+                precomposedForm?.outputs
+                    // create array of metadata objects
+                    .map((formOutput, index) => {
+                        const { label } = formOutput;
+                        // final ordering of outputs differs from order in send form
+                        // outputsPermutation contains mapping from @trezor/utxo-lib outputs to send form outputs
+                        // mapping goes like this: Array<@trezor/utxo-lib index : send form index>
+                        const outputIndex = outputsPermutation.findIndex(
+                            (p: number) => p === index,
+                        );
+                        const metadata: Extract<MetadataAddPayload, { type: 'outputLabel' }> = {
+                            type: 'outputLabel',
+                            entityKey: account.key,
+                            txid, // txid becomes available, use it
+                            outputIndex,
+                            value: label,
+                            defaultValue: '',
+                        };
+                        return metadata;
+                    })
+                    // filter out empty values AFTER creating metadata objects (see outputs mapping above)
+                    .filter(output => output.value)
+                    // propagate metadata to reducers and persistent storage
+                    .forEach((output, index, arr) => {
+                        const isLast = index === arr.length - 1;
+                        dispatch(metadataActions.addAccountMetadata(output, isLast));
+                    });
+            }
+        } else {
+            dispatch(
+                notificationsActions.addToast({
+                    type: 'sign-tx-error',
+                    error: sentTx.payload.error,
+                }),
+            );
+        }
+
+        dispatch(cancelSignTx());
+
+        // resolve sign process
+        return sentTx;
+    };
+
+export const signTransaction =
+    (formValues: FormState, transactionInfo: PrecomposedTransactionFinal) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        const device = selectDevice(getState());
+        const { account, network } = getState().wallet.selectedAccount;
+
+        if (!device || !account) return;
+
+        const enhancedTxInfo: PrecomposedTransactionFinal = {
+            ...transactionInfo,
+            rbf: false,
+        };
+
+        if (account.networkType === 'ethereum') {
+            const definitions = await getEthereumDefinitions({
+                chainId: network?.chainId,
+                contractAddress: enhancedTxInfo.token?.contract.substring(2),
+            });
+            enhancedTxInfo.isTokenKnown = !!definitions?.encoded_token;
+        }
```

Comment:
I do try to rebase every time I create a new PR. But I’ve started working on this branch before `develop` had those changes. 

Done. Though, not sure whether this logic even needed, since you mentioned that the token logic is not related to staking.

https://github.com/everstake/trezor-suite/pull/4/commits/2ed39aaff08c6509b0f48d27cef85971e157b43d

Reply:
Yes, sorry, you can safely remove that

Reply:
https://github.com/everstake/trezor-suite/pull/4/commits/6657f77ea59101128c868e64b771e83d8b95b21f